### PR TITLE
fix: guarantee that max prefix length is < min prefix length + child size

### PR DESCRIPTION
**Pull request by crodriguezvega**

### CODE REVIEW

**Review by AdityaSripal — APPROVED** [review]
Nice work! Ideally we would also have a test that malleates the prefix in order to fool the proof system that now fails. You should be able to pull out the test case from the audit report so don't have to create yourself

File: rust/src/verify.rs [hunk 1]
```diff
@@ -612,19 +623,32 @@ mod tests {
...
-                            valid_inner,
+                            valid_inner.clone(),
                         ],
                     },
                     spec: depth_limited_spec,
                     valid: false,
                 },
             ),
+            (
+                "rejects inner spec with max prefix length >= min prefix lenght + child size",
```
Comment (colin-axner) [inline rust/src/verify.rs:640]:
```suggestion
                "rejects inner spec with max prefix length >= min prefix length + child size",
```

**Review by colin-axner — APPROVED** [review]
Agree that we should have the malleate test case (also a unit test for go code?). The audit recommended updating the proto documentation to indicate this restriction as well

**Comment by colin-axner** [comment]
requesting review from @romac to verify the rust changes

**Comment by colin-axner** [comment]
gonna mark as draft until:
- [ ] malleate test added
- [ ] unit test for check against inner spec
- [ ] proto docs updated
- [ ] rust formatting fixed

**Comment by romac** [comment]
> requesting review from @romac to verify the rust changes

LGTM

File: go/ops.go [hunk 2]
```diff
@@ -166,6 +166,10 @@ func (op *InnerOp) CheckAgainstSpec(spec *ProofSpec, b int) error {
 		return errors.New("spec.InnerSpec.ChildSize must be >= 1")
 	}
 
+	if spec.InnerSpec.MaxPrefixLength >= spec.InnerSpec.MinPrefixLength+spec.InnerSpec.ChildSize {
```
Comment (colin-axner) [inline go/ops.go:169]: for what it is worth. Spec fields being validated against themselves are better off in a spec.Validate() function, but for the sake of minimizing changes, this approach makes sense to me

**Review by damiannolan — APPROVED** [review]

**Pull request merged**